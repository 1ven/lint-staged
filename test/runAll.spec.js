/* eslint no-console: 0 */

import stripColors from 'strip-ansi'
import runAll from '../src/runAll'

let stdout = []
console.log = jest.fn((input) => {
    stdout.push(JSON.stringify(stripColors(input)).replace(/\[\d\d:\d\d:\d\d\]\W/, ''))
    return stdout.join('\n')
})

const packageJson = {
    scripts: {
        mytask: 'echo "Running task"'
    }
}

describe('runAll', () => {
    beforeEach(() => {
        stdout = []
    })

    it('should skip all tasks if there are no staged files', (done) => {
        const config = {
            verbose: true,
            linters: {
                '*': 'mytask'
            }
        }

        runAll(packageJson, config).then(() => {
            expect(stdout).toMatchSnapshot()
            done()
        })
    })

    it('should run tasks on staged files', (done) => {
        const config = {
            '*': 'mytask'
        }

        runAll(packageJson, config).then(() => {
            expect(stdout).toMatchSnapshot()
            done()
        })
    })
})
