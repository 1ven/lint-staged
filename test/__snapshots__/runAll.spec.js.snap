// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runAll should run tasks on staged files 1`] = `
Array [
  "\\"Running lint-staged with the following config:\\\\n\\"",
  "{\\"concurrent\\":true,\\"gitDir\\":\\".\\",\\"linters\\":{\\"*\\":\\"mytask\\"},\\"renderer\\":\\"verbose\\",\\"verbose\\":true}",
  "\\"\\\\n\\"",
  "\\"Running tasks for * [started]\\"",
  "\\"Running tasks for * [skipped]\\"",
  "\\"→ No staged files match *\\"",
]
`;

exports[`runAll should skip all tasks if there are no staged files 1`] = `
Array [
  "\\"Running lint-staged with the following config:\\\\n\\"",
  "{\\"concurrent\\":true,\\"gitDir\\":\\".\\",\\"linters\\":{\\"*\\":\\"mytask\\"},\\"renderer\\":\\"verbose\\",\\"verbose\\":true}",
  "\\"\\\\n\\"",
  "\\"Running tasks for * [started]\\"",
  "\\"Running tasks for * [skipped]\\"",
  "\\"→ No staged files match *\\"",
]
`;
